plugin/trino-open-policy-agent/src/main/java/io/trino/plugin/openpolicyagent/schema/OpaQueryResult.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openpolicyagent.schema;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+
+import static java.util.Objects.requireNonNullElse;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public record OpaQueryResult(@JsonProperty("decision_id") String decisionId, Boolean result)
+{
+    @Override
+    public @NotNull Boolean result()
```

Comment:
I wanted to default the field to `false` if for whatever reason it is not provided on the response from OPA (an "undefined" OPA response without this field may happen in some cases, and we should treat these in the same way as an authorization denied one)

If I make this `boolean` then it clashes with the definition of `Boolean result` on the record definition. That one needs to be the boxed type such that it can be constructed in cases where the JSON response just doesn't contain this field.

To my knowledge (please correct me if I'm wrong), Jackson doesn't have a succinct way of defaulting the value with an annotation, so I either need to:
- Keep this as-is, or
- Use a `JsonSetter` so I can override the value